### PR TITLE
Create Font::ChangeTemplatedText to allow a quick change of a text element.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR follows #1040

### CODE REVIEW

**Comment by LGM-Doyle** [comment]
Rebased.

File: GG/GG/TextControl.h [hunk 1]
```diff
@@ -173,6 +173,12 @@ class GG_API TextControl : public Control
     virtual void SetText(const std::string& str,
                          const std::vector<boost::shared_ptr<Font::TextElement> >&text_elements);
 
+    /** Change TextControl's text to replace the text at templated \p targ_offset with \p new_text.
+
+        This does not reparse the TextControl's text. It is faster than SetText on a new
+        string. It will not find white space in the inserted text. */
+    void ChangeTemplatedText(const std::string& new_text, size_t targ_offset);
```
Comment (geoffthemedio) [inline GG/GG/TextControl.h:201]: Description and name are unclear. Does this insert text and push the following back to make room? Or replace the text of one text element starting immediately before/after the offset? Or replace all text start from the specified offset?

File: GG/GG/Font.h [hunk 2]
```diff
@@ -564,6 +564,17 @@ class GG_API Font
                            std::vector<boost::shared_ptr<TextElement> >& text_elements,
                            std::vector<boost::shared_ptr<TextElement> >::iterator starting_from) const;
 
+    /** Change \p text_elements and \p text to replace the text at \p targ_offset with \p new_text.
+
+        This does not recompute the text_elements. It is faster than ComputeTextElements on a new
+        string. It will not find white space in the inserted text.
+
+        \p text and \p text_elements are assumed to be consistent and both will be changed. */
+    void ChangeTemplatedText(std::string& text,
```
[the remark this answers is not in the document]
Comment (geoffthemedio) [reply GG/GG/Font.h:595]: Why "templated" ?

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

**Review by LGM-Doyle — COMMENTED** [review]
I've tried to expand and clarify the comments for the declarations of both of these functions, including an example.

**Pull request merged**